Full-Stack-Collective/connectient
# Phone number input

## Description

This PR adds a new input for the phone number on the appointment form using react-phone-number-input: https://www.npmjs.com/package/react-phone-number-input?activeTab=code

This package saves the phone number as a string starting with a + and removes all of the spaces/other formatting that are visible to the user when they type in the input box. 

When the page loads there is a large error that pops up in the console - Warning: Failed prop type: The prop `metadata` is marked as required in `ForwardRef(PhoneInputWithCountry)`, but its value is `undefined`. This does not affect anything about the usability of the package. The data saves correctly in the database. This only seems to be an issue when using this package in conjunction with react-hook-form. Someone else is having this same issue and they have flagged it a week ago here: https://gitlab.com/catamphetamine/react-phone-number-input/-/issues/195
I'm going to be keeping my eye on this for whenever someone fixes it and posts a solution. 

## Related Issue

closes #73 

## Acceptance Criteria

This phone number input should visually format the number for the user to make it easier to read while using the form.
The user should be able to select the country for their phone number.
This input should be required and display an error message if it is not filled in.
This input should validate the number to make sure that it is a possible phone number and display an error message if it is not.

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

![Screen Shot 2023-07-24 at 2 33 28 PM](https://github.com/Full-Stack-Collective/connectient/assets/70449557/1dfd5cf1-8700-456e-8281-e45bc579b3a3)

## Testing Steps / QA Criteria

Run npm i in terminal to load in new package. Go to the appointment page and enter a phone number in the input box. The default country is set to Trinidad, but using the drop down menu any country can be selected. If the field is clicked on and then left blank, an error message will pop up. While a person is entering a number, an error message will appear saying the phone number is invalid until a valid number is entered. Once the form is submitted, a correct string of the entered phone number will be sent to the DB.
